framework/src/controller/application.js

```diff
@@ -6,6 +6,10 @@ const version = require('../version');
 const validator = require('./helpers/validator');
 const applicationSchema = require('./schema/application');
 const constantsSchema = require('./schema/constants');
+
+/* As for now we don't support this constants to be configured
+by the user. This will change in the future */
+const nonConfigurableConstants = require('./constants');
```

Comment:
Please rename the file to `non_configureable_constants.js`

Reply:
BTW if rest are `configureable constants` then these are configurations, not constants. So I suggest to move the rest of them to configurations and then later describe which are network configurations and which are node configurations.

Reply:
I agree with you but I also think that then `non_configurable_constants` are indeed constants and the rest as you said are `configurations`.

Reply:
I think constants are always non configurable...=0

Reply:
Agreed. Then we could move `schema/constants.js` to `schema/config.js` and move and rename `non_configurable_constants` to `schema/constants.js`. But we still need to pass them to `global.constants` as it would be too much refactor for the scope of this PR

Reply:
So `schema/config.js` would have what it has now, plus: 

```
module.exports = {
	constants: {
		id: 'constants',
		type: 'object',
		required: [
			'EPOCH_TIME',
			'BLOCK_TIME',
			'MAX_TRANSACTIONS_PER_BLOCK',
			'REWARDS',
		],
		properties: {
			EPOCH_TIME: {
				type: 'string',
				format: 'date-time',
				description:
					'Timestamp indicating the start of Lisk Core (`Date.toISOString()`)',
			},
			BLOCK_TIME: {
				type: 'number',
				min: 1,
				description: 'Slot time interval in seconds',
			},
			MAX_TRANSACTIONS_PER_BLOCK: {
				type: 'integer',
				min: 1,
				description: 'Maximum number of transactions allowed per block',
			},
			REWARDS: {
				$ref: 'rewards',
				default: {
					MILESTONES: [
						'500000000', // Initial Reward
						'400000000', // Milestone 1
						'300000000', // Milestone 2
						'200000000', // Milestone 3
						'100000000', // Milestone 4
					],
					OFFSET: 2160,
					DISTANCE: 3000000,
				},
			},
		},
		additionalProperties: false,
	},
	rewards: {
		id: 'rewards',
		type: 'object',
		required: ['MILESTONES', 'OFFSET', 'DISTANCE'],
		description: 'Object representing LSK rewards milestone',
		properties: {
			MILESTONES: {
				type: 'array',
				items: {
					type: 'string',
					format: 'amount',
				},
				default: [
					'500000000', // Initial Reward
					'400000000', // Milestone 1
					'300000000', // Milestone 2
					'200000000', // Milestone 3
					'100000000', // Milestone 4
				],
				description: 'Initial 5, and decreasing until 1',
			},
			OFFSET: {
				type: 'integer',
				min: 1,
				description: 'Start rewards at block (n)',
			},
			DISTANCE: {
				type: 'integer',
				min: 1,
				description: 'Distance between each milestone',
			},
		},
		additionalProperties: false,
	},
	default: {
		EPOCH_TIME: new Date(Date.UTC(2016, 4, 24, 17, 0, 0, 0)).toISOString(),
		BLOCK_TIME: 10,
		MAX_TRANSACTIONS_PER_BLOCK: 25,
		REWARDS: {
			MILESTONES: [
				'500000000', // Initial Reward
				'400000000', // Milestone 1
				'300000000', // Milestone 2
				'200000000', // Milestone 3
				'100000000', // Milestone 4
			],
			OFFSET: 2160, // Start rewards at first block of the second round
			DISTANCE: 3000000, // Distance between each milestone
		},
	},
};
```

Which are indeed meant to be configured by the user.

Reply:
And `schema/constants.js` which have exactly the same content as the content now present on `non_configurable_constants.js`

Reply:
@shuse2 @nazarhussain @michielmulders please let me know your position on this :)

Reply:
Ok, for me the naming makes more sense now! 👍

Reply:
it looks good but does `rewards` need to be defined outside?

Reply:
@shuse2 I can move it inside, it was outside to avoid nesting but it's the only reference we have